wireapp/wire-ios-mono
# feat: Remove custom self delete timer option - WPB-10861

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10861" title="WPB-10861" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-10861</a>  Remove Custom option in self delete timer
  </td></table>
  <br />
 






Also fix memory link when assigning `didTapViewClosure`




### Issue

On **development builds only** we have the option to select a custom time for the self delete timer both when sending a message or in the self deleting messages group options. 3rd party QA found a collection of bugs related to this.

As this feature is likely not used much this PR removes the feature to avoid future confusion with QA.

Additionally this PR fixes a memory leak due to explicitly capturing `self` when configuring `picker.didTapViewClosure`:

<img width="466" alt="Screenshot 2024-09-09 at 15 15 39" src="https://github.com/user-attachments/assets/d419f1be-7a07-47cd-b567-dc891a3fa9de">



### Testing

1. Go to a group which you have admin permissions.
2. Select the `timer`.
3. Scroll through the picker.
4. Ensure there is no `Custom` options.
5. Select 10 seconds.
6. Send a message.
7. Ensure the message is deleted after 10 seconds.
8. Go to group details > `Self-deleting messages`.
9. Ensure there is no `Custom` option.

---

### Checklist
